File: java/problems/p0070_climbing_stairs/Solution.java
package problems.p0070_climbing_stairs;

class Solution {
    public int climbStairs(int n) {
        if (n == 1)
            return n;

        int a = 1;
        int b = 2;

        for (int i = 3; i <= n; i++) {
            int tmp = a + b;
            a = b;
            b = tmp;
        }
        return b;
    }
}

File: typescript/problems/p0070_climbing_stairs/solution.ts
function climbStairs(n: number): number {
  if (n == 1) return n;

  let a: number = 1;
  let b: number = 2;

  for (let i = 3; i <= n; i++) {
    let tmp = a + b;
    a = b;
    b = tmp;
  }

  return b;
}

export default climbStairs;
